module/Solr/src/Solr/Bridge/ResultConverter.php

```diff
@@ -80,12 +81,20 @@ public function convert(AbstractPaginationQuery $filter, ArrayAccess $response)
         // iterate over Solr response to preserve sorting
         foreach ($response['response']['docs'] as $doc) {
             // check if entity exists
-            if (!isset($entities[$doc->id])) {
-                // skip non-existent entity
-                continue;
+            if (isset($entities[$doc->id])) {
+                // use found entity
+                $entity = $entities[$doc->id];
+            } else {
+                if (!isset($emptyEntity)) {
```

Comment:
Better use dedicated dummy entities. If someone messed up one, it won't affect the others.